src/bin/miri.rs

```diff
@@ -242,21 +242,49 @@ fn run_compiler(
     callbacks: &mut (dyn rustc_driver::Callbacks + Send),
 ) -> ! {
     if target_crate {
-        // Miri needs a custom sysroot for target crates.
-        // If no `--sysroot` is given, the `MIRI_SYSROOT` env var is consulted to find where
-        // that sysroot lives, and that is passed to rustc.
-        let sysroot_flag = "--sysroot";
-        if !args.iter().any(|e| e == sysroot_flag) {
-            // Using the built-in default here would be plain wrong, so we *require*
-            // the env var to make sure things make sense.
-            let miri_sysroot = env::var("MIRI_SYSROOT").unwrap_or_else(|_| {
-                show_error!(
-                    "Miri was invoked in 'target' mode without `MIRI_SYSROOT` or `--sysroot` being set"
-                    )
-            });
+        if env::var_os("MIRI_STD_AWARE_CARGO").is_some() {
```

Comment:
I would strongly prefer if we could avoid having such hacks in the Miri driver.  This should all be taken care of by `cargo-miri`.

`cargo-miri` already injects itself into every invocation from cargo, so it should be entirely possible to do any further command-line argument adjustments there rather than doing them inside the Miri driver.